src/secmlt/optimization/gradient_processing.py

```diff
@@ -79,8 +79,13 @@ def __call__(self, grad: torch.Tensor) -> torch.Tensor:
         NotImplementedError
             Raises NotImplementedError if the norm is not in 2, inf.
         """
-        if self.p == 2:  # noqa: PLR2004
-            return normalize(grad.data, p=self.p, dim=0)
+        if self.p == LpPerturbationModels.get_p(
+            LpPerturbationModels.L1
+        ) or self.p == LpPerturbationModels.get_p(LpPerturbationModels.L2):
+            original_shape = grad.data.shape
+            return normalize(grad.data.flatten(start_dim=1), p=self.p, dim=1).view(
```

Comment:
normalize is just a rescale in the L2 norm in this case, we have to solve the max problem in the Lp ball instead